airbyte-integrations/connectors/source-typeform/source_typeform/source.py

```diff
@@ -245,13 +247,24 @@ def path(self, **kwargs) -> str:
 
 
 class SourceTypeform(AbstractSource):
+    def get_auth(self, config: MutableMapping) -> AuthBase:
+        if config.get("token"):
+            return TokenAuthenticator(token=config["token"])
+        return SingleUseRefreshTokenOauth2Authenticator(
+            config, token_refresh_endpoint="https://api.typeform.com/oauth/token"
+        )
+
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         try:
             form_ids = config.get("form_ids", []).copy()
+            if config.get("credentials"):
```

Comment:
I'd rather replace this piece of code with `self.get_auth(config)` and then calling `.get_auth_headers()` on it

Reply:
thanks, added call to this method